pkg/client/kubernetes/scaling_test.go

```diff
@@ -132,6 +133,41 @@ var _ = Describe("scale", func() {
 			Expect(ScaleStatefulSetAndWaitUntilScaled(context.TODO(), runtimeClient, key, 2)).To(Succeed(), "scale done")
 		})
 	})
+
+	Describe("#WaitUntilNoPodRunningForDeployment", func() {
+		It("should wait until there are no running pods for the deployment", func() {
```

Comment:
Can you also add a test for the negative case - where the function times out if there are still running pods?